### DISCUSSION TITLE
Env Prefix Users

### DISCUSSION THREAD

**Discussion by abrostoff** [littleK0i/SnowDDL#53] [General]
When using the env prefix, what password is set for the env-prefix’s users? I thought maybe it would be the same as the password set in user.yml, but it doesn’t seem to work. Please let me know, thanks!

**Comment by littleK0i** [comment]
It should definitely be the same password.

Here is the place in code which sets passwords for new users: https://github.com/littleK0i/SnowDDL/blob/master/snowddl/resolver/user.py#L50-L53

Please note, SnowDDL cannot "compare" passwords for existing users, since Snowflake does not provide this information in any shape or form. It is only possible to "refresh passwords" (set again forcefully) for all users.

You may do so by adding CLI option `--refresh-user-passwords`.

**Comment by abrostoff** [comment]
We are using that --refresh-user-password option, but after debugging the code, we realize that snowddl isn't running the `create_object` function that you mentioned above. Here is the code that we used to debug:

```
 python  -m trace -l SnowDDL/snowddl/app/base.py  \
  --refresh-user-passwords \
  -c snowddl-admin \
  -a $SNOWFLAKE_ACCOUNT \
  -u $SNOWFLAKE_USERNAME \
  -p $SNOWFLAKE_PASSWORD \
  --env-prefix $ENV_PREFIX \
  --show-sql \
  --clone-table \
  apply
```

**Comment by littleK0i** [comment]
Refresh would run these lines: https://github.com/littleK0i/SnowDDL/blob/master/snowddl/resolver/user.py#L92-L93

Followed by: https://github.com/littleK0i/SnowDDL/blob/master/snowddl/resolver/user.py#L242-L266

Since you have `--show-sql` option, you should be able to see all executed statements.

You may also try to "destroy" and re-"apply" environment with env-prefix to see how users are created from scratch.
